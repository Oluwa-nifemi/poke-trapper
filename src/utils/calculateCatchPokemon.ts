const calculateCatchPokemon = (): boolean => Math.random() > 0.5

export default calculateCatchPokemon