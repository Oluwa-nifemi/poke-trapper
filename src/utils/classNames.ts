/* This function allows you to merge several classes together.
 * For conditional classes the first item in the array should be the condition (boolean)
 */
export const classNames = (...args: (string | [boolean, string])[]) => {
    return args.filter(className => {
        if (Array.isArray(className)) {
            return className[0];
        }

        return className;
    }).map((className) => {
        if (Array.isArray(className)) {
            return className[1];
        }

        return className;
    }).join(' ')
};