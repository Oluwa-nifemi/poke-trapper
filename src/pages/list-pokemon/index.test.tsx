import {render, screen} from "tests/utils";
import ListPokemon from "./index";
import {waitFor} from "@testing-library/react";

describe("Test List Pokemon", () => {
    it("should render correctly", async () => {
        const {container} = render(<ListPokemon/>);
        await waitFor(() => expect(screen.queryByText("Loading...")).toBeNull());
        expect(container).toMatchSnapshot();
    });
});