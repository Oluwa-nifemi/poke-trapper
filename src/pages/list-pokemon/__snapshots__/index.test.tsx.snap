// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List Pokemon should render correctly 1`] = `
<div>
  <h1
    class="header"
  >
    List of pokemon
  </h1>
  <section
    class="pokemonRows"
  >
    <article
      class="pokemonRowsHeader"
    >
      <span>
        Name
      </span>
      <span>
        Owned
      </span>
    </article>
    <article
      class="pokemon"
    >
      <a
        class="pokemonName"
        href="/bulbasaur"
      >
        bulbasaur
      </a>
      <span>
        0
      </span>
    </article>
    <article
      class="pokemon"
    >
      <a
        class="pokemonName"
        href="/ivysaur"
      >
        ivysaur
      </a>
      <span>
        0
      </span>
    </article>
    <article
      class="pokemon"
    >
      <a
        class="pokemonName"
        href="/venusaur"
      >
        venusaur
      </a>
      <span>
        0
      </span>
    </article>
  </section>
</div>
`;
